webapp/niapp.yaml

```diff
@@ -0,0 +1,416 @@
+      responses:
+        200:
+          $ref: '#/responses/CreateWebAppResponse'
+        401:
+          $ref: '#/responses/Unauthorized'
+  '/webapps/{id}/sharing':
+    put:
+      tags:
+        - metadata
+      summary: Update sharing settings of the webapp
+      description: Update sharing settings of the webapp
+      operationId: update-webapp-sharing
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+        - $ref: '#/parameters/UpdateWebAppSharingRequest'
+      responses:
+        200:
+          $ref: '#/responses/UpdateWebAppSharingResponse'
+        400:
+          $ref: '#/responses/ValidationError'
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+  '/webapps/shared-emails':
+    get:
+      tags:
+        - metadata
+      summary: List the emails the user has shared webapps with
+      description: List the emails the user has shared webapps with
+      operationId: list-shared-emails
+      responses:
+        200:
+          $ref: '#/responses/SharedEmailsResponse'
+        401:
+          $ref: '#/responses/Unauthorized'
+  '/webapps/{id}/content':
+    get:
+      tags:
+        - content
+      summary: Get webapp content
+      description: Get the content of a webapp. ContentType varies from JSON for dashboards and templates or redirect to main HTML for WebVIs
+      operationId: get-content-index
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+      responses:
+        200:
+          description: The webapp's content. JSON for Dashboards, HTML for WebVIs
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+    put:
+      tags:
+        - content
+      summary: Upload content for an existing webapp
+      description: Upload content an existing webapp
+      operationId: update-content
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+        - in: body
+          required: true
+          name: content
+          description: The webapp content to upload
+          schema:
+            $ref: '#/definitions/WebAppContent'
+      responses:
+        200:
+          description: Success
+        400:
+          $ref: '#/responses/ValidationError'
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+  '/api/webapps/{id}/content/{path}':
+    get:
+      tags:
+        - content
+      summary: Get webapp content by path
+      description: Get webapp content at path. For webapps with multiple files (e.g. WebVIs)
+      operationId: get-content
+      parameters:
+        - $ref: '#/parameters/WebAppId'
+        - $ref: '#/parameters/WebAppContentPath'
+      responses:
+        200:
+          $ref: '#/definitions/WebAppContent'
+        401:
+          $ref: '#/responses/Unauthorized'
+        404:
+          $ref: '#/responses/NotFound'
+
+  '/webapps/{id}/duplicate/{sourceId}':
```

Comment:
If we're duplicating an existing webapp, it seems like something like `/webapps/{existingWebappId}/duplicate` is a bit more fluent for this RPC call, and then possibly have a name/id/other parameters for the duplicated one passed in the request body. As it's worded now, it seems like `duplicate` is being called on an ID which doesn't yet exist, which may be confusing.

Reply:
This is the documentation of the existing implementation.